src/main/java/io/fabric8/maven/docker/util/WaitUtil.java

```diff
@@ -48,6 +48,26 @@
 
     private WaitUtil() {}
 
+    public static void wait(int wait, Callable<Void> callable) throws ExecutionException {
+        try {
+            FutureTask<Void> task = new FutureTask<>(callable);
+            task.run();
+
+            /*
+             * pad the wait time so that the future will never time out but still complete - if pad == 0,
+             * set to default docker timeout
+             */
+            task.get(((wait == 0) ? 10 : wait) + 5, TimeUnit.SECONDS);
+        } catch (@SuppressWarnings("unused") InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (@SuppressWarnings("unused") TimeoutException e) {
+            /*
+             * this should not happen b/c the wait time has been padded such that if the docker daemon fails
+             * to kill the container in the alloted time, it will be booted out via the ExecutionException
+             */
+        }
```

Comment:
this is the bit i am most interested in feedback on. the idea here is the call to the `Future` itself should never time out b/c we should be relying on the docker daemon to throw the error in the event the container doesn't shut down in time.

the question here is what to do in the event that actually fails, which is theoretically possible. throw an `IllegalStateException`?